Automattic/wp-calypso
# Gutenberg: Simple payment read/save/update

#### Changes proposed in this Pull Request

* Add some simple UI to modify block attributes

* Read, create and update simple payment posts 😵

* This is not intended as a finalized UI, but a step to help with development.

*  Existing wpcom flows are used, they should be swapped for more current Gutenberg approaches in the future.

* There are lots of FIXME and TODO comments, but this works behind the `gutenberg/block/simple-payments` feature flag 👍 

* This works in both local Jetpack's wp-admin Gutenberg as well Calypso-Gutenberg.

* Depends on Jetpack changes to Simple Payments API:
    - https://github.com/Automattic/jetpack/pull/10402
    - D19907-code


## Testing instructions

### Test in Jetpack
* Jurassic Ninja: https://jurassic.ninja/create?jetpack-beta&branch=update/simple-payments-rest-api-meta&gutenpack&gutenberg&calypsobranch=add/gutenpack-simplepayments-edit-ui
* Connect Jetpack, purchase "premium" or "business" plan
* Create new post in Gutenberg and add "Payment button" block
* Open the post and see the button
* Open the post in Calypso classic editor and see the button
* Open the post in Calypso using this branch and Gutenberg and see the button: https://calypso.live/gutenberg/post?branch=add/gutenpack-simplepayments-edit-ui

### Test in Calypso
- Apply patch to your sandbox: D19907-code
- Open https://calypso.live/gutenberg/post?branch=add/gutenpack-simplepayments-edit-ui
- Choose a site with Simple payments enabled (purchase "premium" or "business" plan)

### Manual checking
Apply these patches and cross-test the block in different editors:
- This PR
- https://github.com/Automattic/jetpack/pull/10402
- D19907-code

## Checklist:
* Play with the Simple Payments block and see if attributes update
* A new payment should save a new post. Watch network requests.
* Update and watch it save again.
* The save/update are triggered when the block is deselected or editor's post gets updated
* Save the post and see the page: you should see simple payments button
* Refresh the editor page, note how block pulls in info from API and ignores block `attributes`: you can test this by removing or mocking attributes in the editor's code-editor, saving the post while in code-editor and reloading the page again.
